cni/pkg/install/monitoring.go

```diff
@@ -0,0 +1,47 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package install
+
+import "istio.io/pkg/monitoring"
+
+var (
+	resultLabel                   = monitoring.MustCreateLabel("result")
+	resultSuccess                 = "SUCCESS"
+	resultCopyBinariesFailure     = "COPY_BINARIES_FAILURE"
+	resultReadSAFailure           = "READ_SERVICE_ACCOUNT_FAILURE"
+	resultCreateKubeConfigFailure = "CREATE_KUBECONFIG_FAILURE"
```

Comment:
Do you think these are too granular tags? I am guess these should ~never happen except if things are extremely broken. At that point, it may be more valuable to just diagnose based on logs?

I am not against this, just seems like we should have tags for expected errors, but not sure about unexpected ones?

Reply:
I think this will be useful for the case where log is not so easy to get, like in the vendor env.

I feel it is ok to track these errors at this granularity. If cardinality is totally OK (there aren't many) and the label is not confusing, make label a bit detailed seems handy to me.